### PR TITLE
Split Fortran2008.py into separate files (closes #411)

### LINKED ISSUE #411: split Fortran2008.py into one file per class
The Fortran2008.py file is getting large and it would be better to have a Fortran2008 directory containing a separate file for each class with an __init__ that imports all of the files. There may be an issue with the class generation and class hierarchy generation.

### PR DESCRIPTION
**Pull request by rupertford**
Splits Fortran2008.py into one file per class.

### CODE REVIEW

**Comment by rupertford** [comment]
Ready for first review from @arporter or @sergisiso

File: src/fparser/two/tests/fortran2008/test_submodule_stmt_r1117.py [hunk 1]
```diff
@@ -92,44 +93,32 @@ def test_simple_error6():
...
+    monkeypatch.setattr(splitline, "splitparen", lambda x: ["", "(id", "name"])
     with pytest.raises(NoMatchError) as excinfo:
         dummy_ = Submodule_Stmt("submodule (id name")
     assert "Submodule_Stmt: 'submodule (id name'" in str(excinfo.value)
 
 
 def test_splitparen_error(monkeypatch):
     """Test that if the first argument of the splitparen is not empty then
-    an error is returned.
+    an error is returned. Monkeypatch to force this error.
```
Comment (arporter) [inline src/fparser/two/tests/fortran2008/test_submodule_stmt_r1117.py:118]: This is all a bit worrying. Is it possible to fix splitparen instead?

File: src/fparser/two/Fortran2008/submodule_stmt_r1117.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        return f"SUBMODULE ({self.items[0]}) {self.items[1]}"
+
+    def get_name(self):  # C1114
+        """Fortran 2008 constraint C1114
+        return the submodule name. This is used by the base class to check
+        whether the submodule name matches the name used for the end
+        submodule statement if one is provided.
+
+        :return: the name of the submodule stored in a Name class
+        :return type: :py:class:`fparser.two.Fortran2003.Name`
```
Comment (arporter) [inline src/fparser/two/Fortran2008/submodule_stmt_r1117.py:118]: "rtype" please

File: src/fparser/two/Fortran2008/submodule_stmt_r1117.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        if parent_id_brackets[-1] != ")":
+            return None
+        parent_id = parent_id_brackets[1:-1].lstrip().rstrip()
+        # Format is OK from this Class' perspective. Pass on
+        # parent_identifier and submodule name to appropriate classes
+        return Parent_Identifier(parent_id), Submodule_Name(submodule_name)
+
+    def tostr(self):
+        """return the fortran representation of this object"""
+        # return self.string  # this returns the original code
```
Comment (arporter) [inline src/fparser/two/Fortran2008/submodule_stmt_r1117.py:108]: rm this line?

**Review by arporter — CHANGES_REQUESTED** [review]
I've assumed that all you've done is move code to separate files so have not (well almost) requested changes to any of the content of those files.
However, I'm a bit worried by the `splitparen` shenanigans. Is it possible to get a handle on what's going wrong with that method?

File: src/fparser/two/Fortran2008/submodule_stmt_r1117.py (hunk 2 above)
Comment (rupertford) [reply src/fparser/two/Fortran2008/submodule_stmt_r1117.py:118]: Oops.

File: src/fparser/two/Fortran2008/submodule_stmt_r1117.py (hunk 3 above)
Comment (rupertford) [reply src/fparser/two/Fortran2008/submodule_stmt_r1117.py:108]: Agreed. Removed.

File: src/fparser/two/tests/fortran2008/test_submodule_stmt_r1117.py (hunk 1 above)
Comment (rupertford) [reply src/fparser/two/tests/fortran2008/test_submodule_stmt_r1117.py:118]: This is no different to the tests that were there before splitting out the code, other than a bit of reformatting. I suggest that fixing this issue - if needed - should be done in a separate PR.

Comment (arporter) [reply src/fparser/two/tests/fortran2008/test_submodule_stmt_r1117.py:118]: OK :-)

**Review by arporter — APPROVED** [review]
All looks good now.
Have tested with PSyclone and all fine.
Will proceed to merge.

**Pull request merged**